Task-49284: copy/past problem in chat text area

wassimmhamdi opened the pull request:
Problem: The background color is copied also and since the text color became black the message is unreadable
Fix: The background color shouldn’t be copied

AzmiTouil commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -220,6 +222,34 @@ export default {
+        selection.removeAllRanges();
+        selection.addRange(range);
+      }
+    },
```
What happens if I copy/paste an image ? it works?

wassimmhamdi replied:
yes it works correctly !

Commit:
Task-49284: copy/past problem in chat text area
Problem: The background color is copied also and since the text color became black the message is unreadable
Fix: The background color shouldn’t be copied and the image can be copied

AzmiTouil commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,29 @@ export default {
       return message;
-    }
+    },
+    paste(e) {
+      // consider the first item (can be easily extended for multiple items)
+      const item = e.clipboardData.items[0];
+      //test if the type of item e
+      if (item.type.indexOf('image') === 0)
+      {
+        const blob = item.getAsFile();
+        const reader = new FileReader();
+        reader.onload = function(event) {
+          document.getElementById('container').src = event.target.result;
+        };
+        reader.readAsDataURL(blob);
+      }
+      else {
+        // cancel paste
+        e.preventDefault();
+        // get text representation of clipboard
+        this.text = (e.originalEvent || e).clipboardData.getData('text/plain');
+        // insert text manually
+        document.execCommand('insertHTML', false, this.text);
+      }
+    },
```
Format the code please!

wassimmhamdi replied:
code formatted

AzmiTouil commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,29 @@ export default {
+      //test if the type of item e
+      if (item.type.indexOf('image') === 0)
+      {
+        const blob = item.getAsFile();
```
```suggestion
        const pastedImage = item.getAsFile();
```

wassimmhamdi replied:
variable renamed

AzmiTouil commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,29 @@ export default {
+        reader.onload = function(event) {
+          document.getElementById('container').src = event.target.result;
+        };
+        reader.readAsDataURL(blob);
```
```suggestion
        reader.readAsDataURL(pastedImage);
```

wassimmhamdi replied:
variable renamed

Commit:
fix formatting and change variable name

ahamdi commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,27 @@ export default {
+        const pastedImage = item.getAsFile();
+        const reader = new FileReader();
+        reader.onload = function (event) {
+          document.getElementById('container').src = event.target.result;
```
What is this element **container** created ? is it a hidden element ? I think you need to create one that put the image inside ?

wassimmhamdi replied:
element **container** changed to **messageComposerArea**

Commit:
element container changed to messageComposerArea

rdenarie commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,27 @@ export default {
+        const pastedImage = item.getAsFile();
+        const reader = new FileReader();
+        reader.onload = function (event) {
+          document.getElementById('messageComposerArea').src = event.target.result;
```
I think you should use this.$refs.messageComposerArea
as it is done elsewhere in this component.

wassimmhamdi replied:
changes done !

Commit:
using this.$refs.messageComposerArea instead of messageComposerArea

rdenarie commented on application/src/main/webapp/vue-app/components/ExoChatMessageComposer.vue:
```diff
@@ -347,7 +348,27 @@ export default {
+        // get text representation of clipboard
+        this.text = (e.originalEvent || e).clipboardData.getData('text/plain');
+        // insert text manually
+        document.execCommand('insertHTML', false, this.text);
```
This command seems deprecated : https://developer.mozilla.org/fr/docs/Web/API/Document/execCommand
We should use something else

wassimmhamdi replied:
I change it to other method that do the same work

rdenarie reviewed (changes requested):
****

Commit:
modifications without focus()

Commit:
fix problem of cursor position after copy/paste message

Commit:
fix problem of cursor position after copy/paste message
